### ISSUE TITLE
igraph_all_st_mincuts should not allow zero capacities

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1363]
`igraph_all_st_mincuts` does not complain about zero capacities, but simply drops zero-capacity edges from the result. It should not allow zero capacities.

Example graph with capacities:

![image](https://user-images.githubusercontent.com/1212871/78048306-d3cebd80-7379-11ea-80cf-3ede7d75e031.png)

Asking for all minimum cuts from 1 to 2:

```
In[188]:= IGFindMinimumCuts[g, 1, 2]
Out[188]= {{1 \[DirectedEdge] 2}}
```

Notice that `1 -> 2` is not a cut. `{1 -> 2, 1 -> 3}` would be a cut. But `1->3` has zero capacity and is dropped from the result.

@vtraag  Can you take a look, since you touched this function last time?

**Comment by szhorvat** [comment]
On a related note, we should add a warning to the docs:

This finds all cuts with minimum total capacity. In a weighted graph, there is usually only one, as the weights break the symmetry. Of course, it is possible that some cuts happen to have the same total capacity. But even if it is so, the function will likely not find them all because the total capacities end up being slightly different due to roundoff errors.

In short: if the user is really expecting more than one minimum cut in a weighted graph, it is only safe to use integer weights (as these are unaffected by roundoff errors).

Example:

![image](https://user-images.githubusercontent.com/1212871/78051154-910ee480-737d-11ea-8cdc-3a725f583706.png)

**Comment by vtraag** [comment]
I believe that the whole procedure devised by [Provan and Shier](https://dx.doi.org/10.1007/BF01961544) is based on the assumption that the capacities are strictly positive. It cannot distinguish between an edge that is not present and an edge with zero capacity. The most straightforward solution would be to simply include a check on the capacities. Shall I implement this?

**Comment by szhorvat** [comment]
Please go ahead if you want to.
